src/main.cpp

```diff
@@ -580,7 +580,7 @@ int main(int argc, const char* argv[]) {
         }
     }
 
-    if (sympy_opt) {
+    if (app.got_subcommand(sympy_opt)) {
```

Comment:
This feels bug-prone. If we need to finalize, we should finalize if it was initialized. But we should ask the wrapper if it was initialized, and not rely on `app.got_subcommand(sympy_opt)`.